### ISSUE TITLE
web端私有部署，电脑没有摄像头，怎么看android端的东西

### ISSUE THREAD

**Issue by wujunling** [starrtc/starrtc-webim-demo#1]
web端私有部署，电脑是云，可以实现没有摄像头看到android端的东西吗

**Comment by lordmen99** [comment]
are u success, setup for web app source? i try call videocall and voip from android into my web app source still not working. can u share ur setup

**Comment by elesos** [comment]
@wujunling 收到，下一版加上此功能。

**Comment by lordmen99** [comment]
i mean, how setup web app source, so can connecting with android app. like in existing demo starrtc.

**Comment by elesos** [comment]
@lordmen99 put “starrtc-webim-demo（for im）” or "starrtc-web-demo(for audio and video chat etc.)" source code in your web server , eg.nginx or apache,  then look at "https://github.com/starrtc/starrtc-server" to deploy the server app,  then following the document "https://docs.starrtc.com/zh-cn/docs/web-10.html" .

**Comment by lordmen99** [comment]
i have doing that, here demo web https://www.indofreelancer.id/rtc/ i can get userid, but when i see in wss like this

:

http://prntscr.com/o0qz7z

**Comment by lordmen99** [comment]
for between android app i have success, build it. then i want try between android app and web,but still stuck.

**Comment by elesos** [comment]
@lordmen99 make sure to upgrade your chrome to newest version.

**Comment by elesos** [comment]
@lordmen99 Which country are you from, sir?

**Comment by lordmen99** [comment]
i have use latest chrome, from indonesia. I still cant make working for web app, u have check my demo web?

**Comment by xass** [comment]
@lordmen99, Please wait for our weekend update, we will fix the problem.

**Comment by lordmen99** [comment]
ok,thank you for support

**Comment by longqianxun** [comment]
> ok,thank you for support

Hi，our web demo support both public deployment and private deployment. I 'v seen your web demo, it's seem like you change the code to make it be private mode, but I can not find your changes in the source. 45.77.157.2, do you know this ip address? If you do not change the code, public mode default, it won't be connect that ip, please check if you make the change. 
If you want to use public, do not change any code
If you want to use private, you should get server program in https://github.com/starrtc/starrtc-server, for web support ,make sure you launch the program under the 'web' folder and shutting the firewall down.After that use the code in index.js like this, replace 10.90.7.70 with your server ip
![1](https://user-images.githubusercontent.com/15863536/59347431-300f1500-8d47-11e9-8064-5e70e8c4e511.png)
Last step , web demo must use https, we provide Self-Signed Certificate under the server 'web' folder, after launch the server, before you use the web demo with private, you must access the url https://your server ip:29991 to 29995 in chrome ,that five ports are used for trust the Certificate, if you have the official certificate, just replace ours and forget the last step. 
Finally, you can use private web demo, web and android can connect between same private config.
Good luck

**Comment by lordmen99** [comment]
yes, i still use existing config in web app demo.  i will try that parameter. just copy and paste inside or where must replace it?

**Comment by longqianxun** [comment]
> yes, i still use existing config in web app demo. i will try that parameter. just copy and paste inside or where must replace it?

replace the ip with yours, that's private, for just test, you can use public mode

**Comment by lordmen99** [comment]
45.77.157.2, its my server with already installed starrtc-server. the code layout in javascript like this, not neat

http://prntscr.com/o0vl4c

where i must put code for private mode in that javascript.

**Comment by longqianxun** [comment]
> 45.77.157.2, its my server with already installed starrtc-server. the code layout in javascript like this, not neat
> 
> http://prntscr.com/o0vl4c
> 
> where i must put code for private mode in that javascript.
You should call the api in index.js like this
![1](https://user-images.githubusercontent.com/15863536/59350313-5e442300-8d4e-11e9-99e4-da438cefb662.png)
See the code in index.js, uncomment that code, and replace 10.90.7.70 with yours

**Comment by lordmen99** [comment]
ok,let my try it

**Comment by lordmen99** [comment]
i get this http://prntscr.com/o0w2o1

i try make videocall to web user id

**Comment by longqianxun** [comment]
> i get this http://prntscr.com/o0w2o1
> 
> i try make videocall to web user id

That‘s mean your server are not ready, please check you launch the server under the ‘web’ folder like
![2](https://user-images.githubusercontent.com/15863536/59352181-9ea5a000-8d52-11e9-922e-cfc653471b0d.png)
after launch server ,run netstat -aptn | grep Serv check the ports are listening

**Comment by lordmen99** [comment]
this my port status, http://prntscr.com/o0wdbj

still cant make video call from android to web and opposite

**Comment by lordmen99** [comment]
if i use android app with 2 phone, its working for videocall,voip,live etc.  then i see if try video call its use port 10086. so, if i want try between web to android and android to web must same port or how? confuse :D

**Comment by lordmen99** [comment]
and what mean this = 
 please check you launch the server under the ‘web’ folder like
cause first time after install server, i run start.sh. its correct?

**Comment by lordmen99** [comment]
i have relaunch from web folder, then now get thiss error =
`star_rtc_lib.min.js:1 WebSocket connection to 'wss://45.77.157.2:10087/' failed: Error in connection establishment: net::ERR_CERT_COMMON_NAME_INVALID`

**Comment by lordmen99** [comment]
i have success, setup it. please delete this thread

**Comment by elesos** [comment]
> i have success, setup it. please delete this thread

could you please summarize the deploy procedure to help other people who have the same problem?

**Comment by lordmen99** [comment]
sure,but i want to know this server how many user  can handle in concurent when 1 use broadcasting streaming? and if we use our server ideal about specification for server least need ram , cpu etc. how about that?
